modules/graphdb/templates/07_cluster_setup.sh.tpl

```diff
@@ -127,45 +139,90 @@ if [ "$INSTANCE_ID" == "$${LOWEST_INSTANCE_ID}" ]; then
       echo "Something went wrong! Check the logs."
     fi
   done
-fi
 
-echo "###########################################################"
-echo "#    Changing admin user password and enable security     #"
-echo "###########################################################"
+  # Setting the security should be done on the Leader node only
+  echo "###########################################################"
+  echo "#    Changing admin user password and enable security     #"
+  echo "###########################################################"
 
-is_security_enabled=$(curl -s -X GET \
+  IS_SECURITY_ENABLED=$(curl -s -X GET \
   --header 'Accept: application/json' \
-  -u "admin:$${GRAPHDB_ADMIN_PASSWORD}" \
+  -u "admin:$${GRAPHDB_PASSWORD}" \
   'http://localhost:7200/rest/security')
 
-# Check if GDB security is enabled
-if [[ $is_security_enabled == "true" ]]; then
-  echo "Security is enabled"
-else
-  # Set the admin password
-  SET_PASSWORD=$(
-    curl --location -s -w "%%{http_code}" \
-      --request PATCH 'http://localhost:7200/rest/security/users/admin' \
-      --header 'Content-Type: application/json' \
-      --data "{ \"password\": \"$${GRAPHDB_ADMIN_PASSWORD}\" }"
-  )
-  if [[ "$SET_PASSWORD" == 200 ]]; then
-    echo "Set GraphDB password successfully"
+  # Check if GDB security is enabled
+  if [[ $IS_SECURITY_ENABLED == "true" ]]; then
+    echo "Security is enabled"
   else
-    echo "Failed setting GraphDB password. Please check the logs!"
-  fi
+    # Set the admin password
+    SET_PASSWORD=$(
+      curl --location -s -w "%%{http_code}" \
+        --request PATCH 'http://localhost:7200/rest/security/users/admin' \
+        --header 'Content-Type: application/json' \
+        --data "{ \"password\": \"$${GRAPHDB_PASSWORD}\" }"
+    )
+    if [[ "$SET_PASSWORD" == 200 ]]; then
+      echo "Set GraphDB password successfully"
+    else
+      echo "Failed setting GraphDB password. Please check the logs!"
+    fi
 
-  # Enable the security
-  ENABLED_SECURITY=$(curl -X POST -s -w "%%{http_code}" \
-    --header 'Content-Type: application/json' \
-    --header 'Accept: */*' \
-    -d 'true' 'http://localhost:7200/rest/security')
+    # Enable the security
+    ENABLED_SECURITY=$(curl -X POST -s -w "%%{http_code}" \
+      --header 'Content-Type: application/json' \
+      --header 'Accept: */*' \
+      -d 'true' 'http://localhost:7200/rest/security')
 
-  if [[ "$ENABLED_SECURITY" == 200 ]]; then
-    echo "Enabled GraphDB security successfully"
-  else
-    echo "Failed enabling GraphDB security. Please check the logs!"
+    if [[ "$ENABLED_SECURITY" == 200 ]]; then
+      echo "Enabled GraphDB security successfully"
+    else
+      echo "Failed enabling GraphDB security. Please check the logs!"
+    fi
   fi
+else
+  echo "The current instance: $INSTANCE_ID is not the lowest, skipping cluster creation"
+fi
+
+echo "####################################"
+echo "#    Updating GraphDB password     #"
```

Comment:
Could go into a separate script idk.

Reply:
I would rather not as it will overcomplicate this script, as it requires the currently active password.